### ISSUE TITLE
Fatal error not a git repo 

### ISSUE THREAD

**Issue by StephaneKazmierczak** [andy-5/wslgit#34]
Hi, 

Thank for porting the git command from wsl to windows, I love the idea, because I really really hate to have to install git on windows when it is perfectly working in my wsl.
 
It seems that wslgit.exe worked for a short while then stopped or maybe it never worked, I not so sure anymore :) Anyway,  wsl.exe git --version returns 2.17.0 but inside VSCode I am getting the following error: 

git rev-parse --show-toplevel
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /)

my git repository is still currently inside my wsl file system, should I also move it to the windows side?

**Comment by StephaneKazmierczak** [comment]
Sorry, I was confused, my repo is indeed in the windows side.  
if I click the git logo bottom left, then I am getting "Git: There are no available repositories"
and the git log shows a repeat of the following command 
> git status -z -u
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git rev-parse --show-toplevel
> git config --get commit.template
Open repository: path 
> git status -z -u

**Comment by andy-5** [comment]
This looks similar to #33, see https://github.com/andy-5/wslgit/issues/33#issuecomment-384940612 for a possible solution.

It looks like you have a bind mount so that you can access your windows files inside WSL at `/c/...` in addition to the standard location `/mnt/c/...`. For some reason, git sometimes uses the `/c/...` variant even if the current path is `/mnt/c/...`. However, wslgit cannot translate the `/c/...` paths to Windows paths, as it doesn't know about the bind mount.

The solution is to not setup the bind mount for the execution of wslgit, e.g. by not using an interactive shell for wslgit. In the latest version, this can be done by setting the Windows environment variable `WSLGIT_USE_INTERACTIVE_SHELL` to `false`, see the Readme for more information.

**Comment by StephaneKazmierczak** [comment]
Yes you are right I had to bind /c with /mnt/c in order to make docker works. 
Ok I had `WSLGIT_USE_INTERACTIVE_SHELL=0` but after setting it to `false` then 
the path in the log became `c:\...\` format and it seems to be working, the repo is recognise in VScode. 
I still get some error in the log:
`fatal: not a git repository (or any parent up to mount point /mnt)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set)`

**Comment by StephaneKazmierczak** [comment]
Thank you for the suport and wslgit ! you are awesome !

**Comment by StephaneKazmierczak** [comment]
Hello just to let you know, I am not sure I should open an issue for that but it seems that if I start a shell (bash.exe) then open VSCode in this case wslgit doesn t work, however if I do the oposite VSCode then my shell, it works perfect :)

**Comment by andy-5** [comment]
I haven't checked this, but maybe the `/c` bind mount is created when you first start a shell and then persists also for all further wsl commands?

Starting VS Code first, I guess it works because the full path to the repository is only checked once at the beginning. All subsequent git commands are based on the initial repository path. If `/c` is mounted later on, it therefore doesn't affect git anymore (unless you open a new project in VS Code).

**Comment by iAbadia** [comment]
> If your git repository is inside your WSL filesystem, how do you access it from VS Code in Windows?
> 
> `wslgit.exe` assumes that the git repository is stored on a Windows filesystem, e.g., at `C:\some\path\`, which is available inside WSL at `/mnt/c/some/path/`. Only paths under `/mnt/{DRIVE}` can be translated to valid Windows paths.
> 
> So, yes, you probably should move your repository to the windows side.

I have my repository on the Ubuntu (in my case Debian distro) since it is accesible in:
`C:\Users\$USER\AppData\Local\Packages\TheDebianProject.DebianGNULinux_$SOMEID\LocalState\rootfs`. Is there any way wslgit can work with this path? If I move the whole project to the windows side it will change the permissions for all files to 755 (NTFS...)
